project/ws/app/src/lib/routes/home/components/filter/filter.component.ts

```diff
@@ -0,0 +1,431 @@
+      // this.tpdsSvc.getFilterDataObject.next(this.assigneeFilterObj)
+    }
+
+    if (this.checkboxes) {
+      this.checkboxes.forEach((element: any) => {
+        element['checked'] = false
+      })
+    }
+  }
+
+  clearFilterWhileSearch() {
+    if (this.checkboxes) {
+      this.checkboxes.forEach((element: any) => {
+        element['checked'] = false
+      })
+    }
+  }
+
+  getDesignation() {
+    this.trainingPlanService.getDesignations().subscribe((res: any) => {
+      if (res && res.result && res.result.response) {
+        this.designationList = res.result.response.content
+      }
+
+    })
+  }
+
+  manageSelectedGroup(event: any, group: any) {
+    if (event.checked) {
+      this.groupList.map((grp: any, index: any) => {
+        if (grp && grp.name === group.name) {
+          this.groupList[index]['selected'] = true
+        }
+      })
+      if (group) {
+        group['selected'] = true
+      }
+      this.assigneeFilterObj['group'].push(group.name)
+    } else {
+      if (this.assigneeFilterObj['group'] &&
+        this.assigneeFilterObj['group'].indexOf(group.name) > -1) {
+        const index = this.assigneeFilterObj['group'].findIndex((x: any) => x === group.name)
+        this.assigneeFilterObj['group'].splice(index, 1)
+      }
+      this.groupList.map((grp: any, index: any) => {
+        if (grp && grp.name === group.name) {
+          this.groupList[index]['selected'] = false
+        }
+      })
+      if (group) {
+        group['selected'] = false
+      }
+    }
+  }
+
+  manageSelectedDesignation(event: any, designation: any) {
+    if (event.checked) {
+      this.designationList.map((ditem: any) => {
+        if (ditem && ditem['name'] === designation.name) {
+          ditem['selected'] = true
+        }
+      })
+
+      this.assigneeFilterObj['designation'].push(designation.name)
+    } else {
+      this.designationList.map((ditem: any) => {
+        if (ditem && ditem['name'] === designation.name) {
+          ditem['selected'] = false
+        }
+      })
+      if (this.assigneeFilterObj['designation'] &&
+        this.assigneeFilterObj['designation'].indexOf(designation.name) > -1) {
+        const index = this.assigneeFilterObj['designation'].findIndex((x: any) => x === designation.name)
+        this.assigneeFilterObj['designation'].splice(index, 1)
+      }
+    }
+  }
+
+  resetFilter() {
+    if (this.competencyTypeList) {
+      this.competencyTypeList.map((citem: any) => {
+        if (citem && citem['selected']) {
+          citem['selected'] = false
+        }
+      })
+    }
+    if (this.competencyThemeList) {
+      this.competencyThemeList.map((titem: any) => {
+        if (titem && titem['selected']) {
+          titem['selected'] = false
+        }
+      })
+    }
+    if (this.competencySubThemeList) {
+      this.competencySubThemeList.map((sitem: any) => {
+        if (sitem && sitem['selected']) {
+          sitem['selected'] = false
```

Comment:
please confirm, if you are using duplicate component then code should be revised as per new requirement

Reply:
Ya I have cleaned up the code, just added as I do the integration further I will clean up the code